### DISCUSSION TITLE
filter from array and append to another file

### DISCUSSION THREAD

**Discussion by MarekGajdosik** [mikefarah/yq#712] [Q&A] [answered]
Hello, I am completely lost in this new syntax and it seems to me I need some help.
I have file1.yml and file2.yml and I want to select part from file1 based on value and append it into the end of file2 with the same structure.

Example:
file1.yml
```
a:
  - "b" : "11"
    "c" :  "12"
    "d" :  "13"

  - "b" : "21"
    "c" :  "22"
    "d" :  "23"
```
And I need to select b, c, d where b == "21" and append it to file2.yml into same structure
```
a:
  - "b" : "21"
    "c" :  "22"
    "d" :  "23"
```
I tried following to select what I need (but I get empty response):
```
yq e '. | select (.a[].b == "21")' file1.yml
```
and I have no idea how can I append 
```
    "b" : "21"
    "c" :  "22"
    "d" :  "23"
```
to the end of "a" of file2.yml

**Comment by mikefarah** [comment, marked as answer]
```
yq ea 'select(fi==1).a += (.a[] | select(fi==0 and .b == "21")) |  select(fi==1)' examples/data1.yaml examples/data2.yaml
```

Synopsis:
- Use `ea` (or eval-all) so that both yaml files are read into memory and the expression is run once (by default `yq` runs the expression once per document to lower memory consumption).
- Use select with the fileIndex `fi` operator to filter out nodes by the file index
- We want to append to '.a' of file index 1 with the contents of .a in file index 0, where b==21
- By default, all files are output, but in this case you only want to output fileIndex 1, so pipe it through a filter again

**Reply by MarekGajdosik** [reply]
This helped me a lot.

Thank you
